serviceAccount/ServiceAccount.psm1

```diff
@@ -0,0 +1,1211 @@
+      
+    If (!(($Role -eq "SecondaryComputer") -and ($DBMode -eq "WID"))) 
+    { 
+        Start sqlcmd.exe -ArgumentList "-S $SQLHost -i $env:Temp\ADFSSQLScripts\UpdateServiceSettings.sql -o $env:Temp\ADFSSQLScripts\UpdateServiceSettings.log" -Wait -WindowStyle Hidden | Out-File $LogPath -Append 
+      
+        If (!$?) 
+        { 
+            Write-Host "`tFailed to execute UpdateServiceSettings.sql. Exiting...." -ForegroundColor "red" 
+            ($ElapsedTime.Elapsed.ToString())+" [ERROR]     Failed to execute UpdateServiceSettings.sql" | Out-File $LogPath -Append 
+            Return $false 
+        } 
+    } 
+    Return $true 
+} 
+
+
+
+#Define functions to export
+
+
+function Set-AdfsServiceAccountRule
+{
+    param(
+        [parameter(ValueFromPipeline=$True)]
+        [string[]]$SecondaryServers,
+
+        [parameter(Mandatory=$true)]
+        [string]$ServiceAccount
+    )
+
+    #Validate provided account exists
+    $Account = $ServiceAccount.Split('\') #Input should have been given in the format domain\user
+    $User = $Account[1]
+    $Lookup = Get-ADUser -Filter {Name -eq $User} 
+    if($Lookup -eq $null)
+    {
+        Write-Error "The specified account does not exist"
+    }
+
+
+    #Create rule with new service account
+    $SID = ConvertTo-Sid($ServiceAccount)
+    $ServiceAccountRule = "@RuleName = `"Permit Service Account`"`nexists([Type == `"http://schemas.microsoft.com/ws/2008/06/identity/claims/primarysid`", Value == `"$SID`"])`n=> issue(Type = `"http://schemas.microsoft.com/authorization/claims/permit`", value = `"true`");`n`n"
+    $Properties = Get-AdfsInternalSettings
+    $Properties.PolicyStore.AuthorizationPolicy = $Properties.PolicyStore.AuthorizationPolicy + $ServiceAccountRule
+    $Properties.PolicyStore.AuthorizationPolicyReadOnly = $Properties.PolicyStore.AuthorizationPolicyReadOnly + $ServiceAccountRule
+    Set-AdfsInternalSettings  $Properties | out-null
+
+
+    $doc = new-object Xml
+    $doc.Load("$env:windir\ADFS\Microsoft.IdentityServer.Servicehost.exe.config")
+    $connString = $doc.configuration.'microsoft.identityServer.service'.policystore.connectionString
+
+
+    if((IsWID -ConnectionString $connString) -eq $true)
+    {
+        if($SecondaryServers.Count -eq 0)
+        {
+            Write-Warning("No list of secondary servers was provided. You must ensure a a sync has occurred on all machines before proceeding to change the service account.")
+        }
+
+        #In the case of WID, sync config among all secondary servers
+        foreach($Server in $SecondaryServers)
+        {
+            Invoke-Command -ComputerName $Server -ScriptBlock {
+                $Date = Get-Date 
+                $Duration = (Get-AdfsSyncProperties).PollDuration
+                Set-AdfsSyncProperties -PollDuration 1
+                Start-Sleep 5
+                Set-AdfsSyncProperties -PollDuration $Duration
+            }
+        }
+    }
+
+
+}
+
+function Set-AdfsServiceAccount 
```

Comment:
This module is basically just the original script and can be ignored for the purposes of this review